tests/trainer/logging/test_eval_loop_logging_1_0.py

```diff
@@ -321,3 +321,62 @@ def test_monitor_val_epoch_end(tmpdir):
         checkpoint_callback=checkpoint_callback,
     )
     trainer.fit(model)
+
+def test_log_works_in_validation_callback(tmpdir):
+    """
+    Tests that log can be called within callback
+    """
+    import pytorch_lightning as pl
+
+    os.environ['PL_DEV_DEBUG'] = '1'
+
+    funcs_name = [f for f in dir(callbacks.Callback) if ("val" in f)]
```

Comment:
while this is very nice and concise, this will also be very hard to debug lol.
Can you just create a callback in this method that implements the hooks you want?

```python
class CB(Callback):
  def on_...():
     self.log(...)
  def on_...():
    self.log(...)
```

Reply:
Done ! Yes, you are right. I was trying something 👍